app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt

```diff
@@ -0,0 +1,346 @@
+ */
+@Composable
+fun SyncedTabsList(syncedTabs: List<SyncedTabsListItem>, listener: SyncedTabsView.Listener) {
+    val listState = rememberLazyListState()
+    LazyColumn(
+        state = listState,
+        verticalArrangement = Arrangement.spacedBy(8.dp),
+        modifier = Modifier.fillMaxSize()
+    ) {
+        items(syncedTabs) { syncedTabItem ->
+            when (syncedTabItem) {
+                is SyncedTabsListItem.Device -> SyncedTabsDeviceItem(syncedTabItem.displayName)
+                is SyncedTabsListItem.Error -> SyncedTabsErrorItem(
+                    errorType = syncedTabItem.errorType,
+                    navController = syncedTabItem.navController
+                )
+                is SyncedTabsListItem.NoTabs -> SyncedTabsNoTabsItem()
+                is SyncedTabsListItem.Tab -> {
+                    SyncedTabsTabItem(
+                        tabTitleText = syncedTabItem.displayTitle,
+                        url = syncedTabItem.displayURL,
+                    ) { listener.onTabClicked(syncedTabItem.tab) }
+                }
+            }
+        }
+    }
+}
+
+/**
+ * Text header for sections of synced tabs
+ *
+ * @param deviceName The name of the user's device connected that has synced tabs.
+ */
+@Composable
+fun SyncedTabsDeviceItem(deviceName: String) {
+    PrimaryText(
+        text = deviceName,
+        fontSize = 12.sp,
+        fontFamily = FontFamily(Font(R.font.metropolis_semibold)),
+        letterSpacing = 0.04.sp,
+        maxLines = 1,
+        modifier = Modifier
+            .padding(16.dp, 16.dp, 8.dp, 7.dp)
+            .fillMaxWidth()
+    )
+}
+
+/**
+ * Synced tab list item UI
+ *
+ * @param tabTitleText The tab's display text.
+ * @param url The tab's URL.
+ * @param onClick The click handler when this synced tab is clicked.
+ */
+@Composable
+fun SyncedTabsTabItem(tabTitleText: String, url: String, onClick: () -> Unit) {
+    Column(
+        Modifier.clickable(
+            onClickLabel = tabTitleText,
+            onClick = onClick
+        )
+    ) {
+        PrimaryText(
+            text = tabTitleText,
+            fontSize = 16.sp,
+            maxLines = 1,
+            modifier = Modifier
+                .padding(16.dp, 0.dp)
+                .fillMaxWidth()
+        )
+        SecondaryText(
+            text = url,
+            fontSize = 12.sp,
+            maxLines = 1,
+            modifier = Modifier
+                .padding(PaddingValues(16.dp, 2.dp, 6.dp, 0.dp))
+                .fillMaxWidth()
+        )
+    }
+}
+
+/**
+ * Error UI to show if there is one of the errors outlined in [SyncedTabsView.ErrorType].
+ *
+ * @param errorType [SyncedTabsView.ErrorType] to be displayed after mapping to a string resource.
+ * @param navController [NavController] to call [NavGraphDirections.actionGlobalTurnOnSync()].
+ */
+@Composable
+fun SyncedTabsErrorItem(errorType: SyncedTabsView.ErrorType, navController: NavController? = null) {
```

Comment:
I don't think `SyncedTabsErrorItem` really needs to know about  `NavController `. I think we should expose a optional lambda function instead for the error button `onClick`. 

In general, I don't think we need to expose controller-type classes into a Composable function since we can just add lambda functions for the event handlers (eg, click, long press), and we should also just pass simple objects that tells the Composable function the necessary state it is trying to display.

Reply:
Big sad
![image](https://user-images.githubusercontent.com/87384386/149399212-4177a41a-9b1d-473b-bec0-8b2f11566ece.png)

I've done the other refactoring to take out the controller from `SyncedTabsView.ErrorType` and `SyncedTabsErrorItem` though.

Reply:
I am definitely not expecting us to put the errorSubUI in `SyncedTabsView.ErrorType`.

Reply:
In fact, I think I expected you to keep it the same as before.

Reply:
Per offline discussion, I referenced the wrong data class in my above comment. The NavController has been replaced with an optional wrapper data class, SyncedTabsListItem.ErrorButton, which will contain a String for the button text and a lambda for clicks.